src/openlcb/DccAccyProducer.hxx

```diff
@@ -0,0 +1,177 @@
+/** @copyright
+ * Copyright (c) 2017, Stuart W Baker
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are  permitted provided that the following conditions are met:
+ *
+ *  - Redistributions of source code must retain the above copyright notice,
+ *    this list of conditions and the following disclaimer.
+ *
+ *  - Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @file DccAccyProducer.hxx
+ *
+ * Producer class that represents 2044 consecutive bits out of DCC accessory
+ * control Well-Known Event ID space.
+ *
+ * @author Stuart Baker
+ * @date 17 June 2017
+ */
+
+#ifndef _OPENLCB_DCCACCYPRODUCER_HXX_
+#define _OPENLCB_DCCACCYPRODUCER_HXX_
+
+#include "executor/CallableFlow.hxx"
+#include "openlcb/NonAuthoritativeEventProducer.hxx"
+#include "openlcb/TractionDefs.hxx"
+
+namespace openlcb
+{
+
+/// Request structure used to send requests to the TractionThrottle
+/// class. Contains parametrized reset calls for properly supporting
+/// @ref StateFlowBase::invoke_subflow_and_wait() syntax.
+struct DccAccyProducerInput : public CallableFlowRequestBase
+{
+    /// Possible subflow commands
+    enum Command
+    {
+        CMD_QUERY, ///< state query
+        CMD_SET, ///< state set/change
+    };
+
+    Command cmd; ///< subflow command
+    uint16_t address; ///< DCC accessory address
+    bool value; ///< DCC accessory address value
+
+};
+
+/// DCC accessory address event producer for the Well-Known DCC Accessory range.
+class DccAccyProducer : public CallableFlow<DccAccyProducerInput>,
+                        protected BitRangeNonAuthoritativeEventP
+{
+public:
+    /// DCC accessory activation values
+    enum
+    {
+        NORMAL = false, ///< normal route
+        REVERSE = true, ///< reverse route
+    };
+
+    /// highest possible DCC address supported
+    static constexpr uint16_t MAX_ADDRESS = 2040;
```

Comment:
I've contemplated that as well.  Not sure if we want to support broadcast, or perhaps even make it an option.